openmc/data/multipole.py

```diff
@@ -391,6 +1010,115 @@ def from_hdf5(cls, group_or_filename):
 
         return out
 
+    @classmethod
+    def from_endf(cls, endf_file, vf_options=None, wmp_options=None):
+        """Generate windowed multipole neutron data from an ENDF evaluation.
+
+        Parameters
+        ----------
+        endf_file : str
+            Path to ENDF evaluation
+        vf_options : dict
+            Dictionary of keyword arguments, e.g. {'rtol': 0.001, 'log': True},
+            passed to :func:`openmc.data.multipole._vectfit_nuclide`
+        wmp_options : dict
+            Dictionary of keyword arguments, e.g. {'search': True, 'log': True},
+            passed to :func:`openmc.data.WindowedMultipole.from_multipole`
+
+        Returns
+        -------
+        openmc.data.WindowedMultipole
+            Resonant cross sections represented in the windowed multipole
+            format.
+
+        """
+
+        if vf_options is None:
+            vf_options = {}
+
+        if wmp_options is None:
+            wmp_options = {}
+
+        # generate multipole data from EDNF
+        mp_data = _vectfit_nuclide(endf_file, **vf_options)
+
+        # windowing
+        return cls.from_multipole(mp_data, **wmp_options)
+
+    @classmethod
+    def from_multipole(cls, mp_data, search=False, log=False, **kwargs):
+        """Generate windowed multipole neutron data from multipole data.
+
+        Parameters
+        ----------
+        mp_data : dictionary or str
+            Dictionary or Path to the multipole data stored in a pickle file
+        search : bool, optional
+            Whether to search for optimal window size and curvefit order
+        log : bool or int, optional
+            Whether to print running logs (use int for verbosity control)
+        **kwargs
+            Keyword arguments passed to :func:`openmc.data.multipole._windowing`
+
+        Returns
+        -------
+        openmc.data.WindowedMultipole
+            Resonant cross sections represented in the windowed multipole
+            format.
+
+        """
+
+        if isinstance(mp_data, str):
+            # load multipole data from file
+            with open(mp_data, 'rb') as f:
+                mp_data = pickle.load(f)
+
+        # windowing with specific options
+        if not search:
+            return _windowing(mp_data, log=log, **kwargs)
```

Comment:
Now that `n_cf` is a required argument in `_windowing`, it needs to be passed here.

Reply:
I guess `n_cf` is packed in `kwargs` and will be passed to `_windowing` automatically?
I tested `h1 = openmc.data.WindowedMultipole.from_multipole("./H1/H1_mp.pickle", n_cf=2, n_win=400)` is working while
`h1 = openmc.data.WindowedMultipole.from_multipole("./H1/H1_mp.pickle", n_win=400)` will give a error
`TypeError: _windowing() missing 1 required positional argument: 'n_cf'`

Reply:
Yeah, the use case where I was running into this was:
```Python
openmc.data.WindowedMultipole.from_endf(filename)
```
In which case `n_cf` is not specified anywhere. We need some default value of `n_cf` for this case.